File: HackerRank-Data-Structures/Arrays & Linked Lists/Delete duplicate-value nodes from a sorted linked list.cpp
SinglyLinkedListNode* removeDuplicates(SinglyLinkedListNode* llist) {
    SinglyLinkedListNode* head = llist;
    SinglyLinkedListNode* current = llist;
    llist = llist->next;
    while(llist != nullptr)
    {
        if(current->data == llist->data) {
            llist = llist->next;
            current->next = llist;
            continue;
        }
        current->next = llist;
        current = current->next;
        llist = llist->next;
    }
    return head;
}

File: HackerRank-Data-Structures/Arrays & Linked Lists/Delete duplicate-value nodes from a sorted linked list.py
def removeDuplicates(llist):
    head = llist
    current = llist
    llist = llist.next
    while llist != None :
        if current.data == llist.data:
            llist = llist.next
            current.next = llist
            continue
        current.next = llist
        current = current.next
        llist = llist.next
    return head
